core/include/traccc/fitting/kalman_filter/two_filters_smoother.hpp

```diff
@@ -167,6 +171,10 @@ struct two_filters_smoother {
             return kalman_fitter_status::ERROR_QOP_ZERO;
         }
 
+        if (getter::element(chi2, 0, 0) < 0.f) {
+            return kalman_fitter_status::ERROR_SMOOTHER_CHI2_NEGATIVE;
```

Comment:
This should be ERROR_UPDATER_CHI2_NEGATIVE or ERROR_FILTER_CHI2_NEGATIVE whichever used in `gain_matrix_updater`.